stellarphot/tests/test_core.py

```diff
@@ -115,6 +118,124 @@ def test_base_enhanced_table_from_existing_table():
     assert len(test_base2['dec']) == 1
 
 
+def test_base_enhanced_table_clean():
+    # Check that the clean method exists
+    test_base = BaseEnhancedTable(table_description=test_descript, input_data=testdata)
+    # Add a row so that we can clean something
+    test_base_two = test_base.copy()
+    test_base_two.add_row(test_base[0])
+    test_base_two['ra'][1] = - test_base_two['ra'][1]
+    test_cleaned = test_base_two.clean(ra='>0.0')
+    assert len(test_cleaned) == 1
+    assert test_cleaned == test_base
+
+
+def a_table(masked=False):
+    test_table = Table(
+        [
+            (1, 2, 3),
+            (1, -1, -2)
+        ],
+        names=('a', 'b'),
+        masked=masked)
+    test_table = BaseEnhancedTable(table_description={'a': None, 'b': None},
+                                   input_data=test_table)
+    return test_table
+
+
+def test_bet_clean_criteria_none_removed():
+    """
+    If all rows satisfy the criteria, none should be removed.
+    """
+    inp = a_table()
+    criteria = {'a': '>0'}
+    out = inp.clean(**criteria)
+    assert len(out) == len(inp)
+    assert (out == inp).all()
+
+
+@pytest.mark.parametrize("condition,input_row",
+                            [('>0', 0),
+                             ('=1', 0),
+                             ('>=1', 0),
+                             ('<-1', 2),
+                             ('=-1', 1)])
+def test_bet_clean_criteria_some_removed(condition, input_row):
+    """
+    Try a few filters which remove the second row and check that it is
+    removed.
```

Comment:
Some of these should remove more than just the second row, right?  I think the comment needs an update.